### PR TITLE
Add persistence for TransactionRecord

### LINKED ISSUE #432: Store raw TransactionRecord in database
**Problem**

Auditors require the ability to access the raw `TransactionRecord` bytes, historically, for all transactions, and that data is currently not stored in the DB. This is needed in order to be able to generate a hash of the `TransactionRecord` raw bytes which will be used in state proof validation (retrieval of a state proof based on the hash of a `TransactionRecord`). Storing just the transaction record hash is not sufficient as someone could have maliciously modified that data in the mirror node database.

**Solution**

* Add a new `transaction.transaction_record_bytes` column
* Add a new `hedera.mirror.importer.parser.record.entity.persist.transactionRecordBytes` property with a default of `false`
* Persist the `recordItem.getRecordBytes()` if the property is true

**Alternatives**

Store just the hash of the raw TransactionRecord to avoid excess data in the database.

**Additional Context**

This is an auditor-related requirement coming in c/o @gregscullard. All fields from the TransactionRecord are already stored as separate columns in the database, but regenerating the exact raw bytes via protobuf serialization, in order to create the hash of the TransactionRecord is not guaranteed because protobuf serialization is non-deterministic.

### PR DESCRIPTION
**Pull request by edwin-greene**
**Description**:
The TransactionRecord can now be persisted to the database if the `transactionRecordBytes` property is set to true. (It is set to false by default.)

**Related issue(s)**:

Fixes #432 

**Notes for reviewer**:

**Checklist**

- [x] Documented (Code comments, README, etc.)
- [x] Tested (unit, integration, etc.)

### CODE REVIEW

File: hedera-mirror-importer/src/main/java/com/hedera/mirror/importer/parser/record/entity/EntityProperties.java [hunk 1]
```diff
@@ -98,6 +98,12 @@ public static class PersistProperties {
          */
         private boolean transactionBytes = false;
 
+        /**
+         * If configured the mirror node will store the raw transaction record bytes on the transaction table.
+         * The transaction record bytes include the signatures used for stateproof validation.
```
Comment (steven-sheehy) [inline hedera-mirror-importer/src/main/java/com/hedera/mirror/importer/parser/record/entity/EntityProperties.java:103]: Remove this second line. It is not accurate.

File: hedera-mirror-importer/src/main/resources/db/migration/v1/V1.86.0__add_transaction_record_bytes.sql
Comment (steven-sheehy) [inline hedera-mirror-importer/src/main/resources/db/migration/v1/V1.86.0__add_transaction_record_bytes.sql:1]: 1.85.3

**Review by steven-sheehy — CHANGES_REQUESTED** [review]

File: hedera-mirror-importer/src/main/java/com/hedera/mirror/importer/parser/record/entity/EntityProperties.java (hunk 1 above)
Comment (edwin-greene) [reply hedera-mirror-importer/src/main/java/com/hedera/mirror/importer/parser/record/entity/EntityProperties.java]: Got it, I have removed it.

File: hedera-mirror-importer/src/main/resources/db/migration/v1/V1.86.0__add_transaction_record_bytes.sql
Comment (edwin-greene) [reply hedera-mirror-importer/src/main/resources/db/migration/v1/V1.86.0__add_transaction_record_bytes.sql:1]: Thanks I have renamed it.

**Review by steven-sheehy — APPROVED** [review]
LGTM

**Review by jascks — APPROVED** [review]
LGTM

**Review by xin-hedera — APPROVED** [review]
LGTM

**Pull request merged**